### ISSUE TITLE
Non-empty responses upon successful PUT

### ISSUE THREAD

**Issue by ajberchek** [matrix-org/matrix-doc#1136]
According to the definition of an HTTP PUT in [MDN web docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT), PUT should not return data in the response body upon a successful request.

The Client Server API specifies Response body parameters for the following PUT requests:
- [6.4.1](https://matrix.org/speculator/spec/HEAD/client_server/unstable.html#put-matrix-client-r0-rooms-roomid-state-eventtype-statekey)
- [6.4.2](https://matrix.org/speculator/spec/HEAD/client_server/unstable.html#put-matrix-client-r0-rooms-roomid-state-eventtype)
- [6.4.3](https://matrix.org/speculator/spec/HEAD/client_server/unstable.html#put-matrix-client-r0-rooms-roomid-send-eventtype-txnid)
- [6.5.2.1](https://matrix.org/speculator/spec/HEAD/client_server/unstable.html#put-matrix-client-r0-rooms-roomid-redact-eventid-txnid)

Should these be changed to GETs or POSTs?

**Comment by Half-Shot** [comment]
These would have to be POSTs, due to needing the event_id result. I doubt it's likely to change though.

**Comment by ajberchek** [comment]
Good point, I don't see any issues with 6.5.2.1 being a POST, but for 6.4.[1-3] it looks like the decision to use PUT was due to POST being unsupported on the "state" path.

What if they were kept as PUTs, but a randomized event_id was passed as a parameter in the PUT request? If that event_id isn't already present, then the server will use it as the unique identifier for the event, and if it is present, it will return an error indicating a different id is needed.

Assuming event_ids are large enough, if the client randomly chooses them and ensures it doesn't conflict with any of the ones it has seen locally, then there is a low probability of it colliding with another event_id on the server, unless it is crafted to do so. Even then, the server would simply reject it and send back an error code.

**Comment by richvdh** [comment]
bleh we're hardly alone in having a REST API that returns a body for PUTs.

Is this actually causing a concrete problem?

**Comment by ajberchek** [comment]
This isn't causing any problems in terms of implementation or usability.

I marked this as an issue to bring it to light and strengthen the protocol. However, I don't think this issue is a priority.
